programs/stake/src/stake_instruction.rs

```diff
@@ -64,14 +64,36 @@ declare_process_instruction!(Entrypoint, DEFAULT_COMPUTE_UNITS, |invoke_context|
         Ok(me)
     };
 
+    // The EpochRewards sysvar only exists after the
+    // enable_partitioned_epoch_reward feature is activated. If it exists, check
+    // the `active` field
+    let epoch_rewards_active = invoke_context
+        .get_sysvar_cache()
+        .get_epoch_rewards()
+        .map(|epoch_rewards| epoch_rewards.active)
+        .unwrap_or(false);
+
+    let error_during_epoch_rewards = || {
+        if epoch_rewards_active {
+            Err::<(), InstructionError>(StakeError::EpochRewardsActive.into())
+        } else {
+            Ok(())
+        }
+    };
+
     let signers = instruction_context.get_signers(transaction_context)?;
-    match limited_deserialize(data) {
-        Ok(StakeInstruction::Initialize(authorized, lockup)) => {
+
+    let stake_instruction: StakeInstruction = limited_deserialize(data)?;
+    if !matches!(stake_instruction, StakeInstruction::GetMinimumDelegation) {
+        error_during_epoch_rewards()?;
+    }
```

Comment:
nit (because `error_during_epoch_rewards` is already self-descriptive) but I think this reads more clearly to have the epoch_rewards_active check inline like this
```
if epoch_rewards_active && !matches(..) {
  return Err(..);
}
```

Reply:
Yeah, fair, since we're only using the fn in one place now.